[GLib] Sprinkle WTF_ALLOW_UNSAFE_BUFFER_USAGE_{BEGIN,END} to fix the build with Clang 19

aperezdc opened the pull request:
#### 268ad18a8768c304785624aff632c83cfb4b214f
<pre>
[GLib] Sprinkle WTF_ALLOW_UNSAFE_BUFFER_USAGE_{BEGIN,END} to fix the build with Clang 19
<a href="https://bugs.webkit.org/show_bug.cgi?id=283002">https://bugs.webkit.org/show_bug.cgi?id=283002</a>

Unreviewed. Fix the build for the JSC GLib API with Clang 19.

* Source/JavaScriptCore/API/glib/JSCContext.cpp:
(jsc_context_evaluate_with_source_uri):
(jsc_context_evaluate_in_object):
(jsc_context_check_syntax):
* Source/JavaScriptCore/API/glib/JSCValue.cpp:
(jsc_value_new_string_from_bytes):
* Source/JavaScriptCore/inspector/remote/glib/RemoteInspectorUtils.cpp:
(Inspector::backendCommandsHash):
* Source/WTF/wtf/StringExtras.h:

Canonical link: <a href="https://commits.webkit.org/286497@main">https://commits.webkit.org/286497@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/81d9b52271fc390aa16aaa93ffff6b3b1fbe6ce3

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/76158 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/55188 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/29060 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/80661 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/27425 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/64332 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/3484 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/59717 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/17864 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/79225 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/49618 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/65408 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/40066 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/47014 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/22891 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/25752 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/69337 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/68147 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/23226 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/82122 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/75434 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/3530 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/2292 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/67943 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/3684 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/65376 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/67253 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/11208 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/9324 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/97688 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11783 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/3478 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/21373 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/3501 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/6928 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/124/builds/5259 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/b6f4b4a75a2a097af760068a78c5654ad39e307f. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/36543

webkit-commit-queue commented:
Commit message contains (OOPS!) and no valid reviewer found, blocking PR #36543. Details: [Build #16759](https://ews-build.webkit.org/#/builders/19/builds/16759)

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/81d9b52271fc390aa16aaa93ffff6b3b1fbe6ce3. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/36543

aperezdc committed:
[GLib] Sprinkle WTF_ALLOW_UNSAFE_BUFFER_USAGE_{BEGIN,END} to fix the build with Clang 19
https://bugs.webkit.org/show_bug.cgi?id=283002

Unreviewed. Fix the build for the JSC GLib API with Clang 19.

* Source/JavaScriptCore/API/glib/JSCContext.cpp:
(jsc_context_evaluate_with_source_uri):
(jsc_context_evaluate_in_object):
(jsc_context_check_syntax):
* Source/JavaScriptCore/API/glib/JSCValue.cpp:
(jsc_value_new_string_from_bytes):
* Source/JavaScriptCore/inspector/remote/glib/RemoteInspectorUtils.cpp:
(Inspector::backendCommandsHash):
* Source/WTF/wtf/StringExtras.h:

Canonical link: https://commits.webkit.org/286497@main

webkit-commit-queue commented:
Committed 286497@main (268ad18a8768): <https://commits.webkit.org/286497@main>

Reviewed commits have been landed. Closing PR #36543 and removing active labels.
